tests/dates/test_date_picker.py

```diff
@@ -0,0 +1,183 @@
+            dmc.DatePicker(
+                label="debounce=False",
+                id="debounce-false",
+                debounce=False,
+                minDate="2024-10-01",
+                maxDate="2024-10-31",
+                **kwargs
+            ),
+            dmc.Box(id="out-2000"),
+            dmc.DatePicker(
+                label="debounce=2000",
+                id="debounce-2000",
+                debounce=2000,
+                minDate="2024-10-01",
+                maxDate="2024-10-31",
+                **kwargs
+            ),
+        ]
+    )
+
+    app.layout = dmc.MantineProvider(debounce)
+
+    @app.callback(
+        Output("out-true", "children"),
+        Output("out-false", "children"),
+        Output("out-2000", "children"),
+        Input("debounce-true", "value"),
+        Input("debounce-false", "value"),
+        Input("debounce-2000", "value"),
+    )
+    def update(d_true, d_false, d_2000):
+        return str(d_true), str(d_false), str(d_2000)
+
+    return app
+
+
+# type=default (select one date)
+def test_001dp_datepicker(dash_duo):
+
+    app = make_app()
+
+    dash_duo.start_server(app)
+    # debounce=True
+    # Wait for the app to load
+    dash_duo.wait_for_text_to_equal("#out-true", "None")
+    # enter a value
+    input_elem = dash_duo.find_element("#debounce-true")
+    # focus element to open calendar
+    input_elem.click()
+
+    picker_elem = dash_duo.find_element(".mantine-Popover-dropdown")
+
+    # select a day from the calendar with the keyboard
+    picker_elem.send_keys(Keys.TAB, Keys.TAB, Keys.ENTER, Keys.ESCAPE)
+
+    # verify the output has not been updated because debounce=True
+    dash_duo.wait_for_text_to_equal("#out-true", "None")
+
+    input_elem.send_keys(Keys.TAB)
+    # verify the output has been updated after input loses focus
+    dash_duo.wait_for_text_to_equal("#out-true", "2024-10-01")
+
+    # debounce=False
+    # verify output is blank
+    dash_duo.wait_for_text_to_equal("#out-false", "None")
+    # enter a value
+    input_elem = dash_duo.find_element("#debounce-false")
+    # focus element to open calendar
+    input_elem.click()
+
+
+    time.sleep(.5)
+    picker_elem2 = dash_duo.find_element(".mantine-Popover-dropdown")
+
+    # select a day from the calendar with the keyboard
+    picker_elem2.send_keys(Keys.TAB, Keys.TAB, Keys.ENTER, Keys.ESCAPE, Keys.TAB)
+    # verify the output has been updated without pressing enter or losing focus
+    dash_duo.wait_for_text_to_equal("#out-false", "2024-10-01")
+
+
+    # debounce is an number
+    input_elem = dash_duo.find_element("#debounce-2000")
+    input_elem.click()
+
+    time.sleep(.5)
+    # select a day from the calendar with the keyboard
+    picker_elem3 = dash_duo.find_element(".mantine-Popover-dropdown")
+    picker_elem3.send_keys(Keys.TAB, Keys.TAB, Keys.ENTER)
+
+
+    # initially no change
+    time.sleep(.5)
+    dash_duo.wait_for_text_to_equal("#out-2000", "None", timeout=1)
```

Comment:
Could add the nice `with pytest.raises` pattern you used in https://github.com/snehilvj/dash-mantine-components/pull/407